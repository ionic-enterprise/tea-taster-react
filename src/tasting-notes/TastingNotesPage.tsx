import { useEffect, useState } from 'react';
import {
  IonContent,
  IonFab,
  IonFabButton,
  IonHeader,
  IonIcon,
  IonItem,
  IonItemOption,
  IonItemOptions,
  IonItemSliding,
  IonLabel,
  IonList,
  IonPage,
  IonTitle,
  IonToolbar,
  useIonModal,
} from '@ionic/react';
import { add } from 'ionicons/icons';

import { TastingNoteEditor } from './TastingNoteEditor';
import { useTea } from '../tea/TeaProvider';
import { useTastingNotes } from './useTastingNotes';
import { TastingNote } from '../models';

const TastingNotesPage: React.FC = () => {
  const { notes, refresh, remove } = useTastingNotes();
  const [note, setNote] = useState<TastingNote | undefined>(undefined);
  const { teas } = useTea();
  const [present, dismiss] = useIonModal(TastingNoteEditor, { onDismiss: () => refresh().then(dismiss), teas, note });

  useEffect(() => {
    refresh();
  }, []);

  const handleAddNote = () => {
    setNote(undefined);
    present();
  };

  const handleUpdateNote = (note: TastingNote) => {
    setNote(note);
    present();
  };

  return (
    <IonPage>
      <IonHeader translucent={true}>
        <IonToolbar>
          <IonTitle>Tasting Notes</IonTitle>
        </IonToolbar>
      </IonHeader>
      <IonContent fullscreen className="main-content">
        <IonHeader collapse="condense">
          <IonToolbar>
            <IonTitle size="large">Tasting Notes</IonTitle>
          </IonToolbar>
        </IonHeader>
        <IonFab vertical="bottom" horizontal="end" slot="fixed">
          <IonFabButton onClick={() => handleAddNote()} data-testid="add-note-button">
            <IonIcon icon={add} />
          </IonFabButton>
        </IonFab>
        <IonList>
          {notes
            .sort((a, b) => a.id! - b.id!)
            .map((note) => (
              <IonItemSliding key={note.id}>
                <IonItem button onClick={() => handleUpdateNote(note)}>
                  <IonLabel>
                    <div>{note.brand}</div>
                    <div>{note.name}</div>
                  </IonLabel>
                </IonItem>
                <IonItemOptions>
                  <IonItemOption color="danger" onClick={() => remove(note)}>
                    Delete
                  </IonItemOption>
                </IonItemOptions>
              </IonItemSliding>
            ))}
        </IonList>
      </IonContent>
    </IonPage>
  );
};
export default TastingNotesPage;
