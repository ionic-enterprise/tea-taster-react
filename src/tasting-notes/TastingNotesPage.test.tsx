import { render } from '@testing-library/react';
import TastingNotesPage from './TastingNotesPage';

describe('<TastingNotesPage />', () => {
  it('renders without crashing', () => {
    const { baseElement } = render(<TastingNotesPage />);
    expect(baseElement).toBeDefined();
  });

  it('renders consistently', () => {
    const { asFragment } = render(<TastingNotesPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
