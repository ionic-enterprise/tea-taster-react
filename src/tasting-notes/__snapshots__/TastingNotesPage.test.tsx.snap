// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TastingNotesPage /> > renders consistently 1`] = `
<DocumentFragment>
  <div
    class="ion-page"
  >
    <ion-header
      translucent="true"
    >
      <!---->
      <ion-toolbar>
        <ion-title>
          Tasting Notes
        </ion-title>
      </ion-toolbar>
    </ion-header>
    <ion-content
      class="main-content"
      fullscreen="true"
    >
      <ion-header
        collapse="condense"
      >
        <!---->
        <ion-toolbar>
          <ion-title
            size="large"
          >
            Tasting Notes
          </ion-title>
        </ion-toolbar>
      </ion-header>
      <ion-fab
        horizontal="end"
        slot="fixed"
        vertical="bottom"
      >
        <ion-fab-button
          data-testid="add-note-button"
        >
          <ion-icon
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><path stroke-linecap='round' stroke-linejoin='round' d='M256 112v288M400 256H112' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
          />
        </ion-fab-button>
      </ion-fab>
      <ion-list>
        <ion-item-sliding>
          <ion-item
            button="true"
          >
            <ion-label>
              <div>
                Lipton
              </div>
              <div>
                Green Tea
              </div>
            </ion-label>
          </ion-item>
          <ion-item-options>
            <ion-item-option
              color="danger"
            >
              Delete
            </ion-item-option>
          </ion-item-options>
        </ion-item-sliding>
        <ion-item-sliding>
          <ion-item
            button="true"
          >
            <ion-label>
              <div>
                Rishi
              </div>
              <div>
                Puer Cake
              </div>
            </ion-label>
          </ion-item>
          <ion-item-options>
            <ion-item-option
              color="danger"
            >
              Delete
            </ion-item-option>
          </ion-item-options>
        </ion-item-sliding>
        <ion-item-sliding>
          <ion-item
            button="true"
          >
            <ion-label>
              <div>
                Lipton
              </div>
              <div>
                Yellow Label
              </div>
            </ion-label>
          </ion-item>
          <ion-item-options>
            <ion-item-option
              color="danger"
            >
              Delete
            </ion-item-option>
          </ion-item-options>
        </ion-item-sliding>
      </ion-list>
    </ion-content>
  </div>
</DocumentFragment>
`;
